packages/sit-onyx/src/i18n/index.ts

```diff
@@ -87,11 +79,13 @@ const createI18n = (options: ProvideI18nOptions = {}) => {
    * Current locale.
    * @default "en-US"
    */
-  const locale = computed(() => unref(options?.locale) ?? "en-US");
+  const locale = toRef(options?.locale ?? "en-US");
+  const userT = options.t ? toRef(options.t) : undefined;
 
-  // If the user provided a custom `t` function it should be used instead of the default one
-  if (options.t) {
-    return { t: options.t, locale };
+  // If the user provided a custom `t` function, it is used instead of the default.
+  // Then we also skip the loading and applying of messages.
+  if (userT) {
+    return { t: userT, locale };
```

Comment:
```suggestion
  // If the user provided a custom `t` function, it is used instead of the default.
  // Then we also skip the loading and applying of messages.
  if (options.t) {
    return { t: toRef(options.t), locale };
```

Like this we could get rid of the `userT` variable